src/arduino/localWebServer.ts

```diff
@@ -32,10 +31,24 @@ export default class LocalWebServer {
         this.app.post(url, handler);
     }
 
-    public start(): void {
-        const port = this.server.listen(0).address().port;
-        // tslint:disable-next-line
-        console.log(`Starting express server on port: ${port}`);
-        this._serverPort = port;
+    /**
+     * Start webserver.
+     * If it fails to listen reject will report its error.
+     */
+    public async start() {
+        return new Promise<void>((resolve, reject) => {
```

Comment:
again, this really shouldn't be `async` -- if you're returning a promise without any awaits, just return the promise.